coap-core/src/main/java/com/mbed/coap/packet/CoapPacket.java

```diff
@@ -115,6 +115,15 @@ public InetSocketAddress getRemoteAddress() {
         return remoteAddress;
     }
 
+    public String getRemoteAddrString() {
+        if (remoteAddress != null) {
+            // to ignore scope_id in case of ipv6
+            return remoteAddress.getHostString().split("%")[0].concat(":" + remoteAddress.getPort());
```

Comment:
Should we always remove the scope ID or in special cases (like %0) only? Does it provide any additional information to us?

Reply:
So far it hasn't provided us any value. It only has made log reading/searching a bit annoying at times. And in some cases the scope id hasn't been printed with ipv6 so this would make it consistent through the system.

Reply:
I mean to me the biggest driver of this requirement is to always log ipv6 addresses the same way.